### PR TITLE
Reposync

### PR DESCRIPTION
**Pull request by tpw56j**
This PR solves problems 1-3 described in the #2378.

The short form option -p works in dnf reposync and yum reposync.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
This PR is dependent on #2380 because it includes the changes from it in here. Thus this needs to be squashed to keep the history clean.

**Comment by SchoolGuy** [comment]
Codecov may fail since our coverage is not good enough yet anyway.

**Comment by tpw56j** [comment]
> This PR is dependent on #2380 because it includes the changes from it in here. Thus this needs to be squashed to keep the history clean.

Sorry, I mixed 2 different changes in one branch. I will try to create a PR in a new branch.

**Comment by SchoolGuy** [comment]
@tpw56j Just force push this branch and drop the commits which are included already.

File: cobbler/actions/reposync.py [hunk 1]
```diff
@@ -378,7 +378,10 @@ def rhn_sync(self, repo):
         if has_rpm_list:
             self.logger.warning("warning: --rpm-list is not supported for RHN content")
         rest = repo.mirror[6:]      # everything after rhn://
-        cmd = "%s %s --repo=%s --download_path=%s" % (cmd, self.rflags, pipes.quote(rest), pipes.quote(self.settings.webdir + "/repo_mirror"))
+        
+        # el7:         -p or --download_path
+        # el8, fedora: -p or --download-path
```
Comment (watologo1) [inline cobbler/actions/reposync.py:383]: I guess here (2nd line of the comment) --download-path is not allowed, only -p?

**Comment by tpw56j** [comment]
> @tpw56j Just force push this branch and drop the commits which are included already.

Can you help me find the ability to push a branch and remove commits in the github interface?

I also tried to find a way to delete the entire my fork or sync fork with the main repository and start over, but I could not find it.

And I want to propose additional changes in the reposync.py code that make it possible to use the librepo API, if available, instead of the YUM API. With these changes, the "cobbler reposync" command runs successfully in Fedora 32. Will these changes be interesting or are you planning to completely rewrite the reposync.py code?

File: cobbler/actions/reposync.py (hunk 1 above)
Comment (tpw56j) [reply cobbler/actions/reposync.py:383]: The option -p works in both yum and dnf, so there will be no need to check these 2 cases in the code.

**Comment by SchoolGuy** [comment]
> Can you help me find the ability to push a branch and remove commits in the web github interface?

The web interface I guess has no possibility to force push something. I think when you edit this on your local computer and then use the cli of git or a git gui which supports force pushing this would be the better way.

> I also tried to find a way to delete the entire my fork or sync fork with the main repository and start over, but I could not find it.

So your fork (and this specific branch) is living at https://github.com/tpw56j/cobbler/tree/reposync if you want to clone it to your local computer you can use https://github.com/tpw56j/cobbler.git or if you have set up ssh then it would be better to use git@github.com:tpw56j/cobbler.git
 
> And I want to propose additional changes in the reposync.py code that make it possible to use the librepo API, if available, instead of the YUM API. With these changes, the "cobbler reposync" command runs successfully in Fedora 32. Will these changes be interesting or are you planning to completely rewrite the reposync.py code?

This would be in my eyes a new branch. We are planning to use pulp in the future but before that happens, I would love to be able to use your code. The pulp integration will take a while I think and thus your code improves compability for Cobbler.

If you are a total git newbie I would recommand having the following link as a reference for reading up on Git: https://git-scm.com/book/en/v2

For the very basics about git and github I would recommend your the following tutorial: https://www.youtube.com/watch?v=SWYqp7iY_Tc

For more advanced stuff experience and reading up in the above reference book would be best I think but of course there are a lot of advanced tutorials available also on YT.

File: cobbler/actions/reposync.py (hunk 1 above)
Comment (watologo1) [reply cobbler/actions/reposync.py:383]: Hm, but the comment as it is does not help much?
You would have to add where it only works with -p or am I missing something?

Comment (tpw56j) [reply cobbler/actions/reposync.py:383]: Yes, in the comment, you may not immediately notice the difference between "_" and "-" in --download<x>path. But this leads to a runtime error in one of the cases. -p works in the el7, el8 and fedora.

**Comment by tpw56j** [comment]
> For more advanced stuff experience and reading up in the above reference book would be best I think but of course there are a lot of advanced tutorials available also on YT.

I did the following:
1. git clone git@github.com:tpw56j/cobbler.git
2. git checkout tpw56j-patch-services.py
3. git reset c89a889^ --hard
4. git push -f origin

Comparison of repositories shows that the extra commits is no longer in this branch, but I can see them in the PR. Maybe I missed something?

**Comment by SchoolGuy** [comment]
So now Github is the problem. You can only open a PR for a single branch. This you need to open a new PR for your new branch. At least that is what I would recommend you for now if you are more experienced of course there are ways of circumventing this but that would go too far for now in my eyes.

**Comment by tpw56j** [comment]
> Beside the one comment which looks like a copy and paste issue to me?..

dnf reposync (as in dnf in fedora or el8) uses --download**-**path

running: /usr/bin/dnf reposync --newest-only --delete --config=/var/www/cobbler/repo_mirror/OEL-7-Latest/.origin/OEL-7-Latest.repo --repoid=OEL-7-Latest --download**_**path=/var/www/cobbler/repo_mirror -a x86_64
received on stdout: 
received on stderr: usage: dnf reposync [-c [config file]] [-q] [-v] [--version]
                    [--installroot [path]] [--nodocs] [--noplugins]
                    [--enableplugin [plugin]] [--disableplugin [plugin]]
                    [--releasever RELEASEVER] [--setopt SETOPTS]
--nobest] [-C]
                    [-R [minutes]] [-d [debug level]] [--debugsolver]
                    [--showduplicates] [-e ERRORLEVEL] [--obsoletes]
                    [--rpmverbosity [debug level name]] [-y] [--assumeno]
--repo
--disable] [-x [package]]
                    [--disableexcludes [repo]] [--repofrompath [repo,path]]
                    [--noautoremove] [--nogpgcheck] [--color COLOR]
                    [--refresh] [-4] [-6] [--destdir DESTDIR] [--downloadonly]
                    [--comment COMMENT] [--bugfix] [--enhancement]
                    [--newpackage] [--security] [--advisory ADVISORY]
                    [--bz BUGZILLA] [--cve CVES]
                    [--sec-severity {Critical,Important,Moderate,Low}]
                    [--forcearch ARCH] [-a [arch]] [--delete] [-m]
                    [--download-metadata] [-n] [-p DOWNLOAD_PATH]
                    [--norepopath] [--metadata-path METADATA_PATH] [--source]
                    [--remote-time] [-u]
dnf reposync: error: unrecognized arguments: --download_path=/var/www/cobbler/repo_mirror

File: cobbler/actions/reposync.py (hunk 1 above)
Comment (tpw56j) [reply cobbler/actions/reposync.py:383]: > Beside the one comment which looks like a copy and paste issue to me?..

dnf reposync (as in dnf in fedora or el8) uses --download-path

running: /usr/bin/dnf reposync --newest-only --delete --config=/var/www/cobbler/repo_mirror/OEL-7-Latest/.origin/OEL-7-Latest.repo --repoid=OEL-7-Latest --download_path=/var/www/cobbler/repo_mirror -a x86_64
received on stdout: 
received on stderr: usage: dnf reposync [-c [config file]] [-q] [-v] [--version]
                    [--installroot [path]] [--nodocs] [--noplugins]
                    [--enableplugin [plugin]] [--disableplugin [plugin]]
                    [--releasever RELEASEVER] [--setopt SETOPTS]
--nobest] [-C]
                    [-R [minutes]] [-d [debug level]] [--debugsolver]
                    [--showduplicates] [-e ERRORLEVEL] [--obsoletes]
                    [--rpmverbosity [debug level name]] [-y] [--assumeno]
--repo
--disable] [-x [package]]
                    [--disableexcludes [repo]] [--repofrompath [repo,path]]
                    [--noautoremove] [--nogpgcheck] [--color COLOR]
                    [--refresh] [-4] [-6] [--destdir DESTDIR] [--downloadonly]
                    [--comment COMMENT] [--bugfix] [--enhancement]
                    [--newpackage] [--security] [--advisory ADVISORY]
                    [--bz BUGZILLA] [--cve CVES]
                    [--sec-severity {Critical,Important,Moderate,Low}]
                    [--forcearch ARCH] [-a [arch]] [--delete] [-m]
                    [--download-metadata] [-n] [-p DOWNLOAD_PATH]
                    [--norepopath] [--metadata-path METADATA_PATH] [--source]
                    [--remote-time] [-u]
dnf reposync: error: unrecognized arguments: --download_path=/var/www/cobbler/repo_mirror

**Comment by tpw56j** [comment]
To solve the problem with commits from different branches, I close the PR. I'll start over by looking at the documentation more carefully.

Thanks for your patience.

**Comment by SchoolGuy** [comment]
@tpw56j I am very grateful for your help and I am very happy if I can help you with anything further. If there is anything from our side (Cobbler Community) which we can do to enable people to contribute more easily just drop a note here or write to `cobbler.project@gmail.com`

**Pull request closed without merging**